tgui/packages/tgui/interfaces/BarberingMenu.js

```diff
@@ -0,0 +1,140 @@
+                <Icon name="magnifying-glass" />
+                <Stack.Item mb="10px" grow>
+                  <Input width="100%" onInput={(e, value) => setSearchText(value)} />
+                </Stack.Item>
+              </Stack>
+              <HairPreviewList search_text={searchText}
+                all_hair_names={styles_keys}
+                all_hair_styles={available_styles} />
+            </Stack>
+          </Stack.Item>
+          <Stack.Item position="fixed" width="200px">
+            <PreviewWindow />
+          </Stack.Item>
+        </Stack>
+      </Window.Content>
+    </Window>
+  );
+};
+
+const HairPreviewList = function (props, context) {
+  const { search_text, all_hair_names, all_hair_styles } = props;
+
+  const filtered_list = all_hair_names.filter((x) => x.toLowerCase().includes(search_text.toLowerCase()));
+  return filtered_list.map((hairName) => (<HairPreview
+    key={all_hair_styles[hairName].id}
+    hair_style={all_hair_styles[hairName]}
+    hair_name={hairName} />));
+};
+
+const HairOptions = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { selected_hair_portion } = data;
+
+  return (
+    <Stack justify="space-between" fontSize="15px">
+      <Stack.Item>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "bottom" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "bottom" })}>Bottom Hair</Button.Checkbox>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "middle" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "middle" })}>Middle Hair</Button.Checkbox>
+        <Button.Checkbox mx="20px" checked={selected_hair_portion === "top" ? 1 : 0} onClick={() => act("change_hair_portion", { "new_portion": "top" })}>Top Hair</Button.Checkbox>
+      </Stack.Item>
+      <Stack.Item width="2px">
+        <Box backgroundColor="white" width="2px" height="100%" />
+      </Stack.Item>
+      <Stack.Item shrink>
+        <Button bold color="red" icon="cut" onClick={() => act("do_hair", { "style_id": null })}>Create Wig</Button>
+      </Stack.Item>
+    </Stack>
+  );
+};
+
+const HairPreview = (props, context) => {
+  const { act, data } = useBackend(context);
+  const { hair_style, hair_name } = props;
+  return (
+    <Section width="140px" direction="column" align="center">
+      <Stack inline justify="space-between">
+        <Stack.Item>
+          <Image pixelated width="60px" height="100px" src={`${hair_style["hair_icon"]}`} />
+          <Box width="100%" fontSize="15px" textAlign="center" mb="10px">{hair_name}</Box>
+          <Button color="blue" height="20px" icon="cut" onClick={() => act("do_hair", { "style_id": hair_style["hair_id"] })}>Cut</Button>
+          <Button color="blue" height="20px" icon="eye" onClick={() => act("update_preview", { "action": "new_hair", "style_id": hair_style["hair_id"] })}>Preview</Button>
+        </Stack.Item>
+      </Stack>
```

Comment:
```suggestion
      <Stack vertical>
        <Stack.Item>
          <Image pixelated width="60px" height="100px" src={`${hair_style["hair_icon"]}`} />
        </Stack.Item>
        <Stack.Item>
          <Box fontSize="15px" textAlign="center">{hair_name}</Box>
        </Stack.Item>
        <Stack.Item>
          <Button color="blue" height="20px" icon="cut" onClick={() => act("do_hair", { "style_id": hair_style["hair_id"] })}>Cut</Button>
          <Button color="blue" height="20px" icon="eye" onClick={() => act("update_preview", { "action": "new_hair", "style_id": hair_style["hair_id"] })}>Preview</Button>
        </Stack.Item>
      </Stack>
```
Possibly something like this (untested). Might want to make use of `Flex` if you want tighter spacing.

Reply:
Seems to work nicely, I'll add it in the next commit.